core/coreunix/add.go

```diff
@@ -336,7 +337,10 @@ func AddR(n *core.IpfsNode, root string) (key string, err error) {
 		return "", err
 	}
 
-	f, err := files.NewSerialFile(root, root, false, stat)
+	segs := strings.Split(root, "/")
+	name := segs[len(segs)-1]
```

Comment:
use `name := filepath.Base(root)` instead of string splitting.
Filepath is aware of system specific separators.

Reply:
This probably broke `go test` on windows - https://ci.ipfs.team/blue/organizations/jenkins/IPFS%2Fgo-ipfs/detail/PR-4919/4/pipeline